raft/raft.go

```diff
@@ -434,6 +448,7 @@ func (r *raft) reset(term uint64) {
 		}
 	}
 	r.pendingConf = false
+	r.readOnly = newReadOnly(r.readOnly.option)
```

Comment:
Oh. I misunderstood your original comment. Right now, we treat readIndexRequest as proposals. So client wont really know if it is dropped or not. In etcd, we use simple timeout. We can monitor term to fail faster. If term is changed, but read index response is not returned, we know it is dropped.